avocado/core/job.py

```diff
@@ -581,6 +581,19 @@ def cleanup(self):
         self.__stop_job_logging()
         if self.__remove_tmpdir and os.path.exists(self.tmpdir):
             shutil.rmtree(self.tmpdir)
+        if (getattr(self.args, "dry_run", False) and not
```

Comment:
I like to extract long conditionals like this to a "group of conditionals" implemented as follows:
```python
def cleanup(self):
    ...
    cleanup_conditions = (
        getattr(self.args, "dry_run", False),
        getattr(self.args, "dry_run_no_cleanup", False),
    )
    if all(cleanup_conditions):
        ....
```
Take a look and see if you like it.

Reply:
I love the readability, there's a minor drawback which is that more code *can* get executed.  In the end, I think your approach wins.